packages/framework/src/Model/Product/ProductOnCurrentDomainElasticFacade.php

```diff
@@ -435,4 +436,29 @@ public function getProductsByCategory(Category $category, int $limit, int $offse
         $productsResult = $this->productElasticsearchRepository->getSortedProductsResultByFilterQuery($filterQuery);
         return $productsResult->getHits();
     }
+
+    /**
+     * @param int $productId
+     * @return array
+     */
+    public function getVisibleProductArrayById(int $productId): array
```

Comment:
What do you suggest? Create a single class for it? Something like `ProductElasticsearchFetcher`?

Reply:
Not sure about the name, but in general yes. I think it would be better

Reply:
> Sadly I must agree, we cannot add method here

Actually you can without any regrets... unless it is not added into interface ;)

Reply:
Well I think that is point of these two classes (ProductOnCurrentDomainElasticFacade and ProductOnCurrentDomainFacade) that they include same public methods, because you can load them via interface. I am not sure, if it is ok to add public methods only in one of them.

Reply:
When you are dependent on interface it make sense to implement method into all implementations and add it to interface. But here and now I was really specific about the implementation, as it does not make sence to me to allow any other than elasticsearch approach in such class (`ProductOnCurrentDomainElasticFacade`).

Anyway do not worry it has been chaged. Just want to point out it is not as bad as your feelings are about it.